src/ashley/machina_extensions/forum_permission/handler.py

```diff
@@ -76,3 +79,14 @@ def get_readable_forums(self, forums, user):
                 )
             )
         return readable_forums
+
+    # pylint:disable = W0201
+    def _get_all_forums(self):
+        """ Returns all forums accessible for the LTIContext of the user. """
+        if not hasattr(self, "_all_forums"):
+            super()._get_all_forums()
+            if self.current_lti_context_id:
+                self._all_forums = Forum.objects.filter(
+                    lti_contexts__id=self.current_lti_context_id
+                )
```

Comment:
I propose to improve:
- don't make the query twice 
- return a list like the parent method does
```suggestion
            if self.current_lti_context_id:
                self._all_forums = list(
                    Forum.objects.filter(
                        lti_contexts__id=self.current_lti_context_id
                    )
                )
            else:
                super()._get_all_forums()
```